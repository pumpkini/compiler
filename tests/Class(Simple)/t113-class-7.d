
class A {
    void a(Z z) {
        Print("entering method a of A");
        z.c();
        Print("exiting method a of A");
    }
}

class Z {
    void c() {
        Print("inside method c of Z");
    }
}

int main() {
    A a;
    Z z;
    a = new A;
    z = new Z;

    a.a(z);
}
