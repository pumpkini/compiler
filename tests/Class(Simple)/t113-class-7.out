entering method a of A
inside method c of Z
exiting method a of A
